lib/HttpSignatureStrategy.js

```diff
@@ -81,8 +82,14 @@ Strategy.prototype.authenticate = function(req) {
     getPublicKey: ['checkRequest', function(callback, results) {
       var publicKey = {id: results.parseRequest.keyId};
       brKey.getPublicKey(
-        publicKey, function(err, publicKey) {
-          callback(err, publicKey);
+        publicKey, function(err, publicKeyFound) {
+          if(err && err.name === 'NotFound' &&
+            bedrock.config.passport.strategies.publicKeyDidLookup) {
+            return getDidPublicKey(publicKey.id, function(err, key) {
```

Comment:
We should generalize this to check the config for the URL scheme and then call into a helper that can call into other helpers based on that URL scheme in order to dereference the URL using the appropriate protocol.